src/org/traccar/CoordinatesHandler.java

```diff
@@ -39,10 +39,11 @@ private Position getLastPosition(long deviceId) {
     @Override
     protected Position handlePosition(Position position) {
         Position last = getLastPosition(position.getDeviceId());
-        if (last != null) {
+        if (last != null && last.getLatitude() != 0 && last.getLongitude() != 0) {
             double distance = DistanceCalculator.distance(
                     position.getLatitude(), position.getLongitude(), last.getLatitude(), last.getLongitude());
-            if (distance < coordinatesMinError || distance > coordinatesMaxError) {
+            if (coordinatesMinError != 0 && distance < coordinatesMinError
+                    || coordinatesMaxError != 0 && distance > coordinatesMaxError && !position.getValid()) {
```

Comment:
I know, but it looks like a bug. I think it's better to make it symmetrical so to say. It won't hurt.